### PR TITLE
Implement autocomplete appearance on select multiple

### LINKED ISSUE #1373: Implement autocomplete appearance on search multiple
As described in [this forum thread](https://forum.opendatakit.org/t/filter-select-multiple-choices-as-per-select-one-search/9149). All conversation about how the feature should look and feel should continue there.

In particular, note that the `search` appearance can't be used yet because there is a conflict with `search()` so this should only use `autocomplete`.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #1373 

#### What has been done to verify that this works as intended?
I used this form which contains all affected widgets: 
[select.txt](https://github.com/opendatakit/collect/files/1277334/select.txt)

I tested also cascading select.

#### Are there any risks to merging this code? If so, what are they?
Refactoring is always risky :smiley:

### CODE REVIEW

**Comment by lognaturel** [comment]
Thanks, @grzesiek2010! For the risks, I think it's helpful to think of the worst case and write a little bit about it. It's the process of thinking through what could go wrong that's helpful. For example, what kind of widgets/forms could be affected? If it's only forms with selects, that's worth mentioning. What about selects from external data sets? Did the issue mention any special cases? In this case, the issue did mention "In particular, note that the `search` appearance can't be used yet because there is a conflict with `search()` so this should only use `autocomplete`". By including the `search` appearance, `search()` has been essentially deprecated. See [here](https://forum.opendatakit.org/t/use-of-two-appearances/9443/3?u=ln) for more details.

@nap2000 can you please verify that this is workable for you? The behavior with the `autocomplete` appearance is the same as for select ones -- it allows for filtering based on a case-insensitive substring. It doesn't do anything fancy with showing selected items separately. I think that should probably be a separate appearance because it will look quite different and may not work on all screen sizes in a reasonable way.

**Comment by grzesiek2010** [comment]
> For the risks, I think it's helpful to think of the worst case and write a little bit about it. It's the process of thinking through what could go wrong that's helpful. For example, what kind of widgets/forms could be affected? If it's only forms with selects, that's worth mentioning. What about selects from external data sets?

I added a form with all affected widgets. Additionally I tested cascading select, external itemsets and data preloading so in this case I think the risk is small and even if there is something wrong I'm sure @mmarciniak90 finds it as it's not complicated.

**Comment by nap2000** [comment]
@lognaturel this approach will be useful. I agree with your comment that showing a list of currently sellected could be an alternative appearance

**Comment by lognaturel** [comment]
Thanks, @nap2000!

@grzesiek2010 "I tested cascading select, external itemsets and data preloading" is great info to add, thanks! @mmarciniak90, it's all yours.

**Comment by mmarciniak90** [comment]
Tested with success on Android 4.1, 4.4, 5.1, 6.0, 7.0
User is able to search on multi select question. 
This functionality works the same like searching on single select question.
Verified also external itemsets and data preloading and I did not notice regression.
screenshot_2017-09-07-13-33-26
screenshot_2017-09-07-13-33-00

File: collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java [hunk 1]
```diff
@@ -206,6 +206,8 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context
                     questionWidget = new ListMultiWidget(context, fep, true);
                 } else if (appearance.startsWith("label")) {
                     questionWidget = new LabelWidget(context, fep);
+                } else if (appearance.contains("search") || appearance.contains("autocomplete")) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java]: This should ONLY use `autocomplete` for now and not `search` to maintain the `search()` appearance function until we deprecate it.

**Comment by lognaturel** [comment]
Thanks @mmarciniak90! What Android version are you using that gives you those black highlights? They should be the blue highlight color, I think. Could you please file an issue for that so we can try to make it consistent?

@jknightco it occurs to me that this is going to conflict with your improvements to the Widgets hierarchy in #1433 and others. Could you please review, see if it matches your sense of how this functionality should be organized and then we can discuss what order to do merges in?

**Comment by heyjamesknight** [comment]
@lognaturel luckily I haven't touched any of these widgets yet—I'm still working my way through the "IBinaryWidget" implementations. Feel free to merge these and I'll rebase before I touch them.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Fantastic! Thanks, @grzesiek2010, @mmarciniak90, @jknightco!

**Comment by mmarciniak90** [comment]
@lognaturel Black highlights are visible on Android 4.1, 4.2, 4.4.
I created an issue #1454 to fix this inconsistency.

**Comment by nap2000** [comment]
@grzesiek2010 Thanks for doing this. Works great.  The request that prompted this came from doctors selecting from 50+ prescriptions, they will be happy (I hope).

**Comment by lognaturel** [comment]
Thanks for taking the time to share that, @nap2000! It's always nice to know how improvements are being used. 

If they're not happy (I hope they'll be!), let's brainstorm what else we could do to help.

**Pull request merged**